testsuites/CBLTester/System_test_multiple_device/test_system_testing.py

```diff
@@ -0,0 +1,283 @@
+            assert sorted(ids) == sorted(x)
+            doc_ids.update(ids)
+    else:
+        # getting doc ids from the dbs
+        _check_doc_count(db_obj_list, cbl_db_list)
+        count = db_obj_list[0].getCount(cbl_db_list[0])
+        itr_count = count / query_limit
+        if itr_count == 0:
+            itr_count = 1
+        for _ in range(itr_count):
+            existing_docs = db_obj_list[0].getDocIds(cbl_db_list[0], query_limit, query_offset)
+            doc_ids.update(existing_docs)
+            query_offset += query_limit
+        print "{} Docs in DB".format(len(doc_ids))
+        query_offset = 0
+        try:
+            # Precautionary creation of user
+            sg_client.create_user(sg_admin_url, sg_db, username, password, channels=channels_sg)
+        except:
+            pass
+    # Configure replication with push_pull for all db
+    replicator_obj_list = []
+    replicator_list = []
+    for base_url, cbl_db, query in zip(base_url_list, cbl_db_list, query_obj_list):
+        repl_obj = Replication(base_url)
+        replicator_obj_list.append(repl_obj)
+        session, _, repl = repl_obj.create_session_configure_replicate(
+            base_url, sg_admin_url, sg_db, username, password, channels_sg, sg_client, cbl_db, sg_blip_url, continuous=True)
+        replicator_list.append(repl)
+        query.query_get_docs_limit_offset(cbl_db, limit=query_limit, offset=query_offset)
+
+    current_time = datetime.now()
+    running_time = current_time + timedelta(minutes=up_time)
+
+    _check_doc_count(db_obj_list, cbl_db_list)
+    x = 1
+    while(running_time - current_time > timedelta(0)):
+
+        print '*' * 20
+        print "Starting iteration no. {} of system testing".format(x)
+        print '*' * 20
+        x += 1
+        ######################################
+        # Checking for doc update on SG side #
+        ######################################
+        docs_to_update = random.sample(doc_ids, num_of_docs_to_update)
+        sg_docs = sg_client.get_all_docs(url=sg_url, db=sg_db, auth=session)["rows"]
+        sg_docs = [doc for doc in sg_docs if doc["id"] in docs_to_update]
+        print "Updating {} docs on SG - {}".format(len(docs_to_update),
+                                                   docs_to_update)
+        sg_client.update_docs(url=sg_url, db=sg_db, docs=sg_docs,
+                              number_updates=num_of_updates, auth=session, channels=channels_sg)
+
+        # Waiting until replicator finishes on all dbs
+        for repl_obj, repl, cbl_db, query in zip(replicator_obj_list,
+                                                 replicator_list,
+                                                 cbl_db_list,
+                                                 query_obj_list):
+            t = Thread(target=_replicaton_status_check, args=(repl_obj, repl))
```

Comment:
What is the reason for thread here. Are you trying to do multi threading with sg update, cbl updates , delete docs from sg and delete docs from cbl? Did you try using Threadpool? You can refere to tests which written in test_no_conflicts.cbl which contains 'concurrenctly' in the test name